mailmerge/__main__.py

```diff
@@ -108,37 +119,42 @@ def main(sample, dry_run, limit, no_limit, resume,
 
     # Run
     message_num = 1 + start
+    recipients = []
     try:
+        log = MailmergeLog(log_path)
         template_message = TemplateMessage(template_path)
         csv_database = read_csv_database(database_path)
         sendmail_client = SendmailClient(config_path, dry_run)
         for _, row in enumerate_range(csv_database, start, stop):
-            sender, recipients, message = template_message.render(row)
-            sendmail_client.sendmail(sender, recipients, message)
-            print_bright_white_on_cyan(
-                ">>> message {message_num}"
-                .format(message_num=message_num),
-                output_format,
-            )
-            print_message(message, output_format)
-            print_bright_white_on_cyan(
-                ">>> message {message_num} sent"
-                .format(message_num=message_num),
-                output_format,
-            )
+            try:
```

Comment:
I think this logic can be simplified after merging in the latest upstream develop.  That includes a try-except block in the message send part.